text/metrics/0146-metrics-prototype-scenarios.md

```diff
@@ -0,0 +1,220 @@
+
+#### Process CPU Usage
+
+Note: the **Host Name** should leverage [`OpenTelemetry
+Resource`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md),
+so it should be covered by the metrics SDK rather than API, and strictly
+speaking it is not considered as a "dimension" from the SDK perspective.
+
+| Host Name | Process ID | CPU% [0.0, 100.0] |
+| --------- | ---------- | ----------------- |
+| MachineA  | 1234       | 15.3              |
+
+#### System CPU Usage
+
+| Host Name | CPU% [0, 100] |
+| --------- | ------------- |
+| MachineA  | 30            |
+
+#### Server Room Temperature
+
+| Host Name | Temperature (F) |
+| --------- | --------------- |
+| MachineA  | 65.3            |
+
+### Push Metrics
+
+These are the push metrics - the value is reported (via the API) when it is
+available, and collected (via the SDK) based on the ask from consumer(s). If
+there is no ask from the consumer, the API will be no-op and the data will be
+dropped on the floor.
+
+#### Received HTTP Requests
+
+Note: the **Client Type** is passed in via the [`OpenTelemetry
+Baggage`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/baggage/api.md),
+strictly speaking it is not part of the metrics API, but it is considered as a
+"dimension" from the metrics SDK perspective.
+
+| Host Name | Process ID | Client Type | HTTP Method | HTTP Host | HTTP Flavor | Peer IP   | Peer Port | Host IP   | Host Port |
+| --------- | ---------- | ----------- | ----------- | --------- | ----------- | --------- | --------- | --------- | --------- |
+| MachineA  | 1234       | Android     | GET         | otel.org  | 1.1         | 127.0.0.1 | 51327     | 127.0.0.1 | 80        |
+| MachineA  | 1234       | Android     | POST        | otel.org  | 1.1         | 127.0.0.1 | 51328     | 127.0.0.1 | 80        |
+| MachineA  | 1234       | iOS         | PUT         | otel.org  | 1.1         | 127.0.0.1 | 51329     | 127.0.0.1 | 80        |
+
+#### HTTP Server Duration
+
+Note: the server duration is only available for **finished HTTP requests**.
+
+| Host Name | Process ID | Client Type | HTTP Method | HTTP Host | HTTP Status Code | HTTP Flavor | Peer IP   | Peer Port | Host IP   | Host Port | Duration (ms) |
+| --------- | ---------- | ----------- | ----------- | --------- | ---------------- | ----------- | --------- | --------- | --------- | --------- | ------------- |
+| MachineA  | 1234       | Android     | GET         | otel.org  | 200              | 1.1         | 127.0.0.1 | 51327     | 127.0.0.1 | 80        | 8.5           |
+| MachineA  | 1234       | Android     | POST        | otel.org  | 304              | 1.1         | 127.0.0.1 | 51328     | 127.0.0.1 | 80        | 100.0         |
+
+### Application Requirements
+
+The application owner (developer Y) would only want the following metrics:
+
+* [System CPU Usage](#system-cpu-usage) reported every 5 seconds
```

Comment:
I assume we need to create multiple "pipelines" so that each pipeline can be configured individually.  This implies IMHO that measurements reported from API needs to be routed to each individual pipelines.

Currently, I think this is an issue with our use of a "Default" provider in Library X, if Library X does not take a dependency on the SDK.